source/include/FreeRTOSIPConfigDefaults.h

```diff
@@ -60,6 +60,33 @@
  * MACROS details :
  * https://www.freertos.org/FreeRTOS-Plus/FreeRTOS_Plus_TCP/TCP_IP_Configuration.html
  */
+
+/*---------------------------------------------------------------------------*/
+
+/*
+ * Compile time assertion with zero runtime effects.
+ * It will assert on 'e' not being zero, as it tries to divide by it.
+ */
+
+#ifdef static_assert
+    #define STATIC_ASSERT( e )    static_assert( e, "FreeRTOS-Plus-TCP Error" )
+#elif defined( _Static_assert )
+    #define STATIC_ASSERT( e )    _Static_assert( e, "FreeRTOS-Plus-TCP Error" )
+#else
+/* MISRA Ref 20.10.1 [Lack of sizeof operator and compile time error checking] */
```

Comment:
NOTE: This was previously defined in [BufferAllocation_2.c#L68](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/V4.0.0/source/portable/NetworkInterface/pic32mzef/BufferAllocation_2.c#L68)

So it's just clean-up to move it to here so it can be used in other files

Reply:
Yeah, in order to check the compile time value of any macros that including a casting, you have to use a static assert rather check using the standard method of `#if (ipconfigValue != 0)`. This will allow all the config values to be checked at compilation in FreeRTOSConfigDefaults.h. For example `configMINIMAL_STACK_SIZE` is cast to `( unsigned short )` so you can't use the check `#if (ipconfigIP_TASK_STACK_SIZE_WORDS < configMINIMAL_STACK_SIZE)` but you can use `STATIC_ASSERT(ipconfigIP_TASK_STACK_SIZE_WORDS >= configMINIMAL_STACK_SIZE);`

Reply:
Thanks for the explanation!
I just wanted to add that comment so anybody else who looks at this PR could see that right away 😄